chrysom/listenerClient_test.go

```diff
@@ -32,10 +33,117 @@ var (
 	)
 )
 
+func TestValidateListenerOptions(t *testing.T) {
+	type testCase struct {
+		Description    string
+		ValidateOption ListenerOption
+		Client         ListenerClient
+	}
+
+	tcs := []testCase{
```

Comment:
These tests seem a bit overly explicit.  I'd probably favor just passing an array of options and confirming the New(opts...) returns an error or not.  That makes the code less brittle later.